pmd-java/src/main/java/net/sourceforge/pmd/lang/java/symboltable/VariableNameDeclaration.java

```diff
@@ -105,7 +105,8 @@ private TypeNode getTypeNode() {
         if (isPrimitiveType()) {
             return (TypeNode) getAccessNodeParent().getFirstChildOfType(ASTType.class).jjtGetChild(0);
         }
-        if (!isTypeInferred()) {
+        if (!isTypeInferred()
+            && getAccessNodeParent().getFirstChildOfType(ASTType.class).jjtGetNumChildren() > 0) {
```

Comment:
this seems outdated now... `ASTType` can't be empty

Reply:
Good catch, yes - ASTType can't be empty anymore.